apps/dashboard/app/controllers/projects_controller.rb

```diff
@@ -35,12 +35,10 @@ def update
   def create
     @project = Project.new(project_params)
 
-    # saving .ondemand in project_dataroot 
-    if @project.valid? && @project.save
+    if @project.valid? && @project.save(project_params)
       redirect_to projects_path, notice: 'Project successfully created!'
     else
-      project_error = @project.errors.where(:dir).last
-      redirect_to new_project_path, alert: project_error.message
+      redirect_to new_project_path, alert: @project.errors[:dir].last
```

Comment:
I don't know without looking if this is better.

```suggestion
      redirect_to new_project_path, alert: @project.errors.first
```

Reply:
This would return a hash and not the message.